app/test/prepare_proposal_test.go

```diff
@@ -34,15 +36,15 @@ func TestPrepareProposal(t *testing.T) {
 
 	firstNS := []byte{2, 2, 2, 2, 2, 2, 2, 2}
 	firstMessage := bytes.Repeat([]byte{4}, 512)
-	firstRawTx := generateRawTx(t, encCfg.TxConfig, firstNS, firstMessage, signer, 2, 4, 8, 16)
+	firstRawTx := generateRawTx(t, encCfg.TxConfig, firstNS, firstMessage, signer, allSquareSizes...)
```

Comment:
`AllSquareSizes` is defined [here](https://github.com/rootulp/celestia-app/blob/13076a62c567f624b2508752b7e129467cd71bf9/x/payment/types/square_sizes.go#L26). It is possible to use it here. I opted to retain existing behavior to avoid bloating this PR with a refactor. Happy to update in this PR (if using `AllSquareSizes` is blocking) or in a in a follow-up PR (if non-blocking).

**Note**: I don't see a corresponding `square_sizes_test.go` but will create an issue for it. I think it's important to unit test the behavior of `AllSquareSizes` if we're going to update all tests to using it.

Reply:
> I opted to retain existing behavior to avoid bloating this PR with a refactor

> I don't see a corresponding square_sizes_test.go but will create an issue for it. I think it's important to unit test the behavior of AllSquareSizes if we're going to update all tests to using it.

I understand not wanting to use it if there are no unit tests, so I'm not going to block on this PR. I do however think it falls under the scope of #620. Since we are adding a new validity rule that relies on a function to calculate square sizes. It seems reasonable to also use that same function to generate tests.

if not, pls create an issue with the current context before merging. Thanks!

Reply:
> It seems reasonable to also use that same function to generate tests.

Agreed. Resolved in https://github.com/celestiaorg/celestia-app/pull/666/commits/e5f55a043174b324b20bc2df04f6fdfad72490c0